test/pkg/cluster/cluster.go

```diff
@@ -109,34 +81,53 @@ func AddNFTSvcToNodeDisruptionPolicy(cs *client.ClientSet) error {
 	return nil
 }
 
-func ApplyMachineConfig(yamlInput []byte, c *client.ClientSet) error {
+func ApplyMachineConfigAndCheckChange(yamlInput []byte, c *client.ClientSet) (bool, error) {
 	obj := &machineconfigurationv1.MachineConfig{}
 	if err := yaml.Unmarshal(yamlInput, obj); err != nil {
-		return fmt.Errorf("failed to unmarshal YAML: %w", err)
+		return false, fmt.Errorf("failed to unmarshal YAML: %w", err)
 	}
+	modifiedConfig := obj.Spec.Config
+	var updated bool
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), c, obj, func() error {
+		obj.Spec.Config = modifiedConfig
+		return nil
+	})
 
-	err := c.Create(context.TODO(), obj)
 	if err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create MachineConfig: %w", err)
-		}
+		return false, fmt.Errorf("failed to apply MachineConfig: %w", err)
+	}
 
-		// If it already exists, retrieve the current version to update
-		existingMC := &machineconfigurationv1.MachineConfig{}
-		if err = c.Get(context.TODO(), k8sTypes.NamespacedName{Name: obj.Name}, existingMC); err != nil {
-			return fmt.Errorf("failed to get existing MachineConfig: %w", err)
+	switch op {
```

Comment:
You don't need the switch, return false if unchanged, else return true by default